decoder/vaapiDecoderJPEG_unittest.cpp

```diff
@@ -139,24 +403,41 @@ VAAPIDECODER_JPEG_TEST(Factory)
     doFactoryTest(mimeTypes);
 }
 
-VAAPIDECODER_JPEG_TEST(Decode_Simple)
+template <std::size_t N>
+void decodeJpeg(const std::tr1::array<uint8_t, N>& data, uint32_t fourcc)
 {
     VaapiDecoderJPEG decoder;
     VideoConfigBuffer config;
     VideoDecodeBuffer buffer;
 
     config.flag = 0;
 
-    buffer.data = const_cast<uint8_t*>(g_SimpleJPEG.data());
-    buffer.size = g_SimpleJPEG.size();
+    buffer.data = const_cast<uint8_t*>(data.data());
+    buffer.size = data.size();
     buffer.timeStamp = 0;
 
     ASSERT_EQ(YAMI_SUCCESS, decoder.start(&config));
     ASSERT_EQ(YAMI_DECODE_FORMAT_CHANGE, decoder.decode(&buffer));
     ASSERT_EQ(YAMI_SUCCESS, decoder.decode(&buffer));
     ASSERT_EQ(YAMI_SUCCESS, decoder.decode(&buffer));
 
-    EXPECT_TRUE(decoder.getOutput());
+    SharedPtr<VideoFrame> output = decoder.getOutput();
+    ASSERT_TRUE(output);
+    EXPECT_EQ(fourcc, output->fourcc);
+}
+
+VAAPIDECODER_JPEG_TEST(Decode_Simple)
+{
+    decodeJpeg(g_SimpleJPEG, YAMI_FOURCC_444P);
+    decodeJpeg(g_JPEG420, YAMI_FOURCC_IMC3);
+    /* this will trigger bug in dirver,
+     * see https://bugs.freedesktop.org/show_bug.cgi?id=97633c for details
+     */
+    /*
+    decodeJpeg(g_JPEG422H, YAMI_FOURCC_422H);
+    decodeJpeg(g_JPEG422V, YAMI_FOURCC_422V);
```

Comment:
Instead of generating the JPEG's with ffmpeg (since driver can't handle the output), try using ImageMagick's `convert` program.  It seems to generate images that our driver can handle.  You can convert to different formats with:

`$ convert simple-444P.jpeg -sampling-factor 2x2,2x1,2x1 simple-422V.jpeg`
`$ convert simple-444P.jpeg -sampling-factor 2x2,1x2,1x2 simple-422H.jpeg`

... also you could start with the g_SimpleJPEG image (444P) to generate the others with `convert`.  Finally, I use `okteta` to export the bytes to a c array.

Here's a simple program to write a byte array to a file (in case you need it):
```
#include <stdint.h>
#include <fstream>
#include <tr1/array>

const static std::tr1::array<uint8_t, 844> bytes = { /* put bytes here and set array size appropriately */ };

int main(int argc, char** argv)
{
    std::ofstream of("writebytes.out", std::ios::out | std::ios::binary);
    of.write(reinterpret_cast<const char*>(bytes.data()), bytes.size());
    of.close();
    return 0;
}
```